### PR TITLE
HRCPP-340 Additional SNI and XSite tests

### PR DESCRIPTION
**Pull request by alanfx**
Add SiteManualSwitchTest to JniTest
Add two more scenarios to SimpleTLS-SNI.cpp

### CODE REVIEW

File: test/data/standalone-hotrod-ssl.xml [hunk 1]
```diff
@@ -144,7 +144,7 @@
                 </drivers>
             </datasources>
         </subsystem>
-        <subsystem xmlns="urn:infinispan:server:core:9.0" default-cache-container="local">
+        <subsystem xmlns="urn:infinispan:server:core:8.3" default-cache-container="local">
```
Comment (rigazilla) [inline test/data/standalone-hotrod-ssl.xml]: This should be 9.0 since we run against 9.0.0 server

File: test/data/standalone-hotrod-ssl.xml [hunk 2]
```diff
@@ -158,12 +158,13 @@
                 <local-cache name="namedCache" start="EAGER"/>
             </cache-container>
         </subsystem>
-        <subsystem xmlns="urn:infinispan:server:endpoint:9.0">
+        <subsystem xmlns="urn:infinispan:server:endpoint:8.0">
```
Comment (rigazilla) [inline test/data/standalone-hotrod-ssl.xml]: Should be 9.0 for 9.0.0 server

**Comment by rigazilla** [comment]
@alanfx can you adopt the suggested changes so I'll proceed with the merge?

**Comment by alanfx** [comment]
Sorry, @rigazilla I have updated the PR

**Comment by rigazilla** [comment]
I'll merge this after #262. Now the test fails because windows CI agent trusts sni4 certificate

**Comment by rigazilla** [comment]
merged, thanks!

**Pull request merged**